### ISSUE TITLE
The miniapp ams-example segfaults with 10 number of elements

### ISSUE THREAD

**Issue by ggeorgakoudis** [llnl/AMS#22]
Running as
```
LIBAMS_VERBOSITY_LEVEL=-1 /p/vast1/ggeorgak/projects/ams/AMS/build_ruby/examples/ams_example --precision single --uqtype deltauq-mean -db ./db -S /p/vast1/ggeorgak/projects/ams/AMS/tests/tuple-single.torchscript -e 10
```

Segfaults with output:
```
--> cycle: 1
 material 0: using sparse packing for -5 elems
[ Workflow ] Entering Evaluate with problem dimensions [(-320, 2, -320, 4)]
[ Workflow ] Memory usage at Start is VM:1.37981e+06 RS:141504
[ ResourceManager ] Requesting to allocate -320 values using allocator :mmp-host-quickpool
terminate called after throwing an instance of 'umpire::util::Exception'
  what():  ! Umpire Exception [/var/tmp/blake14/spack-stage/spack-stage-umpire-2022.03.1-zzq5wck6qqis42cbreyywthqbw2vcb2j/spack-src/src/umpire/alloc/MallocAllocator.hpp:43]:  allocate malloc( bytes = 18446744073709551312 ) failed
    Backtrace: 10 frames
    0 0x1555550c6a07 No dladdr: /p/vast1/ggeorgak/projects/ams/AMS/build_ruby/src/libAMS.so(_ZN6umpire5alloc15MallocAllocator8allocateEm+0x527) [0x1555550c6a07]
    1 0x1555550c744e No dladdr: /p/vast1/ggeorgak/projects/ams/AMS/build_ruby/src/libAMS.so(_ZN6umpire8resource21DefaultMemoryResourceINS_5alloc15MallocAllocatorEE8allocateEm+0x4e) [0x1555550c744e]
    2 0x1555550ad940 No dladdr: /p/vast1/ggeorgak/projects/ams/AMS/build_ruby/src/libAMS.so(_ZN6umpire8strategy6mixins17AlignedAllocation16aligned_allocateEm+0x20) [0x1555550ad940]
    3 0x15555505f036 No dladdr: /p/vast1/ggeorgak/projects/ams/AMS/build_ruby/src/libAMS.so(+0x3b036) [0x15555505f036]
    4 0x40cd44 No dladdr: /p/vast1/ggeorgak/projects/ams/AMS/build_ruby/examples/ams_example(_ZN6umpire9Allocator8allocateEm+0x304) [0x40cd44]
    5 0x15555506c355 No dladdr: /p/vast1/ggeorgak/projects/ams/AMS/build_ruby/src/libAMS.so(_ZN3ams15ResourceManager8allocateIbEEPT_m15AMSResourceType+0x65) [0x15555506c355]
    6 0x15555507cd2d No dladdr: /p/vast1/ggeorgak/projects/ams/AMS/build_ruby/src/libAMS.so(_ZN3ams11AMSWorkflowIfE8evaluateEPviPPKfPPfiii+0x1cd) [0x15555507cd2d]
    7 0x407976 No dladdr: /p/vast1/ggeorgak/projects/ams/AMS/build_ruby/examples/ams_example() [0x407976]
    8 0x15554ac1dd85 No dladdr: /lib64/libc.so.6(__libc_start_main+0xe5) [0x15554ac1dd85]
    9 0x4097ee No dladdr: /p/vast1/ggeorgak/projects/ams/AMS/build_ruby/examples/ams_example() [0x4097ee]
```

Debugger (lldb):
```
(lldb) 
frame #9: 0x000015555507cd2d libAMS.so`ams::AMSWorkflow<float>::evaluate(this=0x000000000227fcd0, probDescr=0x000000000226fd00, totalElements=-320, inputs=<unavailable>, outputs=<unavailable>, inputDim=2, outputDim=4, Comm=1140850688) at workflow.hpp:333:65
   330        return;
   331      }
   332      // The predicate with which we will split the data on a later step
-> 333      bool *p_ml_acceptable = ams::ResourceManager::allocate<bool>(totalElements);
   334 
   335      // -------------------------------------------------------------
   336      // STEP 1: call the hdcache to look at input uncertainties
(lldb) p totalElements
(const int) $0 = -320
(lldb) up
frame #10: 0x0000000000407976 ams_example`main(argc=<unavailable>, argv=<unavailable>) at main.cpp:636:32
   633 
   634  #ifdef USE_AMS
   635  #ifdef __ENABLE_MPI__
-> 636            AMSDistributedExecute(workflow[mat_idx],
   637                                  MPI_COMM_WORLD,
   638                                  static_cast<void *>(eoses[mat_idx]),
   639                                  num_elems_for_mat * num_qpts,
(lldb) 
frame #11: 0x000015554ac1dd85 libc.so.6`__libc_start_main + 229
libc.so.6`__libc_start_main:
->  0x15554ac1dd85 <+229>: movl   %eax, %edi
    0x15554ac1dd87 <+231>: callq  0x15554ac34380            ; exit
    0x15554ac1dd8c <+236>: movq   0x8(%rsp), %rax
    0x15554ac1dd91 <+241>: leaq   0x14b07d(%rip), %rdi
```

**Comment by ggeorgakoudis** [comment]
This is fixed after PR #24 fixed precision setting in the example mini-app.

**Comment by ggeorgakoudis** [comment]
This is fixed after PR https://github.com/LLNL/AMS/pull/24 fixed precision setting in the example mini-app.
